### DISCUSSION TITLE
What is the correct tagging for table cells that span multiple rows?

### DISCUSSION THREAD

**Discussion by mbertucci47** [latex3/tagging-project#180] [Q&A]
Here's an example that compares the tagging structure tree of the kernel `\multicolumn` with the [multirow](https://www.ctan.org/pkg/multirow) package's `\multirow`.
```tex
\DocumentMetadata
  {
    lang=en-US,
    pdfversion=2.0,
    pdfstandard=ua-2,
    testphase={phase-III,math,title,table,firstaid}
  }
\documentclass{article}
\usepackage{multirow}

\begin{document}

\begin{tabular}{cc}
\multicolumn{2}{c}{AAA} \\
C & D
\end{tabular}

\begin{tabular}{cc}
\multirow{2}{*}{AAA} & B \\
& D
\end{tabular}

\end{document}
```
<img width="250" alt="multirow_struc" src="https://github.com/user-attachments/assets/24be03bb-210b-4147-8f0e-c2bc48bea21c">

As you can see, `\multicolumn` gives a single cell for the first row, while `\multirow` produces two cells: in the first row, the content, and in the second row, an empty cell.

For a properly tagged table, what is the correct tagging for a cell than spans multiple rows? Should it match the behavior of `\multicolumn` but "transposed", or is the current tagging correct? My guess is that the current behavior is correct because otherwise there is no way to tell which column the "D" in the example should appear in.

P.S. I'm aware no attempt has been made thus far to make `\multirow` tagging correct. I am interested in whether the "out of the box" behavior is correct.

**Comment by zauguin** [comment]
The table model in PDF tagging mostly follows HTML here, so the correct behavior is similar to how this would work in HTML: The `\multirow` only produces one cell in the first row, this cell contains a `RowSpan` attribute specifying that it spans two rows. The second row contains only the `D` column. The placement of `D` in the second column is implicitly given based on the context that the first column is already occupied (by the column from the first row).

**Reply by mbertucci47** [reply]
Thanks @zauguin. So should the `\multicolumn` also have an analogous `ColSpan` attribute? (Also, how does one inspect such attributes? It's not clear to me how to do this from the Acrobat structure tree.)

**Reply by teepeemm** [reply]
Two years later, so you may know by now, but if you right click on the TD and select Properties, you can "Edit Tag..." or "Edit Attribute Classes..." to see that the multicolumn causes a "colspan-2" class.  (Might require Adobe Pro?)
